internal/txapp/txapp.go

```diff
@@ -205,6 +208,33 @@ func (r *TxApp) ChainInfo(ctx context.Context) (int64, []byte, error) {
 	return height, appHash, nil
 }
 
+// ReloadDB reloads the database state into the engine.
+func (r *TxApp) ReloadDB(ctx context.Context) error {
+	tx, err := r.Database.BeginReadTx(ctx)
```

Comment:
Super minor, but I feel like `ReloadCache` or `ReloadMemory` is more descriptive. When I first saw `ReloadDB`, I was very confused what that meant, and the implications it could have on data. `ReloadCache` is much more innocuous (which this is).